jobs/integration/test_service_endpoints.py

```diff
@@ -18,11 +18,15 @@ async def is_pod_running():
 
     try:
         phase = pod["items"][0]["status"]["phase"]
+        _ = pod["items"][0]["spec"]["containers"][0]["env"]
     except IndexError:
         # Pod is not created yet
         return False
+    except KeyError:
+        # Updating Pod envvars
+        return False
 
-    if "Running" in phase:
+    if "Running" in phase and len(pod["items"]) == 1:
```

Comment:
maybe it's best to test something like this

```python
SVC_PORT = 50000
    ...
    try:
        phase = ...
        port = pod["items"][0]["spec"]["containers"][0]["env"]["??PORT??"]
    except (IndexError, KeyError):
        # Pod not created correctly
        return False
        
    if "Running" in phase and str(SVC_PORT) == port:
    ...
```

Reply:
I like the suggested change to check the port, but we need to keep the `len(pod["items"]) == 1` check too. It's needed to prevent a situation where there exists a `pod["items"][1]` that is listening on the wrong port and causing connections to fail.

Reply:
Thanks for the comments and suggestions @addyess @Cynerva. I'll include your suggestions and update the PR.